features/enlinkd/persistence/api/src/main/java/org/opennms/netmgt/enlinkd/model/NodeTopologyEntity.java

```diff
@@ -28,70 +28,141 @@
+import org.opennms.netmgt.model.OnmsIpInterface;
 import org.opennms.netmgt.model.OnmsNode;
-import org.opennms.netmgt.model.monitoringLocations.OnmsMonitoringLocation;
-
-/**
- * This is NOT a Hibernate/JPA entity but rather a lightweight model without less attributes than OnmsNode and no lazy
- * loading. We use it to retrieve node information from the database fast.
- */
-public class NodeTopologyEntity implements Serializable {
-
-    private Integer id;
-    private OnmsNode.NodeType type;
-    private String sysObjectId;
-    private String label;
-    private String location;
-
-    public NodeTopologyEntity(Integer nodeid, OnmsNode.NodeType nodetype, String nodesysoid, String nodelabel, String location){
-        this.id = nodeid;
-        this.type = nodetype;
-        this.sysObjectId = nodesysoid;
-        this.label = nodelabel;
-        this.location = location;
+
+public class NodeTopologyEntity {
+
+    public static NodeTopologyEntity create(OnmsNode node) {
+        if (node == null) {
+            return null;
+        }
+        if (node.getPrimaryInterface() != null) {
+            return new NodeTopologyEntity(node.getId(), 
+                               node.getLabel(),
+                               node.getPrimaryInterface().getIpAddress(),
+                               node.getSysObjectId(), 
+                               node.getSysName(),
+                               node.getLocation() == null ? null : node.getLocation().getLocationName(),
+                               node.getType(),
+                               node.getPrimaryInterface().isManaged());
+        }
+        OnmsIpInterface elected = null;
+        for (OnmsIpInterface ipinterface: node.getIpInterfaces()) {
+            if (elected == null) {
+                elected = ipinterface;
+                continue;
+            }
+            if (!elected.isManaged() && ipinterface.isManaged()) {
+                elected = ipinterface;
+                continue;
+            }
+        }
+        if (elected != null) {
+            return new NodeTopologyEntity(node.getId(), 
+                               node.getLabel(),
+                               elected.getIpAddress(),
+                               node.getSysObjectId(), node.getSysName(),node.getLocation() == null ? null : node.getLocation().getLocationName(),
+                               node.getType(),
+                               elected.isManaged());
+        }
+        return new NodeTopologyEntity(node.getId(), 
+                           node.getLabel(),
+                           null,
+                           node.getSysObjectId(), 
+                           node.getSysName(),
+                           node.getLocation() == null ? null : node.getLocation().getLocationName(),
+                           node.getType(),
+                           false);
+
     }
+    private final int m_nodeId;
+    private final InetAddress m_primaryAddr;
```

Comment:
patrick the InetAddress is required because we use it in topology-map and also because I iuse the same object in EnhancedLinkd to walk devices.

I think that join into database are usually very good and to simplify code we can use a vies instead of using hibernate templates...in this manner we can also annotate the object with @Entity 

Relational Database usually have good join algorithms.